master/internal/stream/system.go

```diff
@@ -325,6 +325,30 @@ func (ps *PublisherSet) Websocket(
 	}
 }
 
+// Websocket is an Echo websocket endpoint.
+func (ps *PublisherSet) Websocket(
+	ssupCtx context.Context,
+	socket *websocket.Conn,
+	c echo.Context,
+) error {
+	reqCtx := c.Request().Context()
+	detCtx, ok := c.(*detContext.DetContext)
+	if !ok {
+		log.Errorf("unable to run PublisherSet: expected DetContext but received %t", reflect.TypeOf(c))
+	}
+	user := detCtx.MustGetUser()
+	return ps.entrypoint(ssupCtx, reqCtx, user, &WrappedWebsocket{Conn: socket}, prepareWebsocket)
+}
+
+func prepareWebsocket(msg stream.PreparableMessage) interface{} {
+	if _, ok := msg.(stream.UpsertMsg); ok {
+		return prepareMessageWithCache(msg, nil)
+	} else if deleted, ok := msg.(stream.DeleteMsg); ok {
+		return newDeletedMsg(deleted.Key, deleted.Deleted)
```

Comment:
It's not clear to me why the prepare function needs to handle different cases.

Why can't prepareWebsocket just call websocket.PrepareMessage() and call it a day?

(not a rhetorical question, I'm having trouble getting all the types right in my head)

Reply:
I looked at it again and I think the right move is to json marshal both and it should work. I think before I made some of these changes, there were differences in handling that I built into these functions, but after a bunch of changes I think it's safe to just json marshal